staging/src/k8s.io/cli-runtime/pkg/genericclioptions/builder_flags.go

```diff
@@ -42,12 +42,14 @@ type ResourceBuilderFlags struct {
 // NewResourceBuilderFlags returns a default ResourceBuilderFlags
 func NewResourceBuilderFlags() *ResourceBuilderFlags {
 	filenames := []string{}
+	kustomize := ""
 
 	return &ResourceBuilderFlags{
 		FileNameFlags: &FileNameFlags{
 			Usage:     "identifying the resource.",
 			Filenames: &filenames,
-			Recursive: boolPtr(true),
+			Kustomize: &kustomize,
+			Recursive: boolPtr(false),
```

Comment:
I'm not saying I disagree that this is how it *should* be, but changing the default on this could cause some problems for existing users who have scripts or CI/CD... it would no longer recurse by default.

I haven't been around here long enough to know whether this is acceptable or not, but I would like to know others opinions about this change.

Reply:
Yes. It may need more validations, just for me I didn't find any reasons for the default value to be 'true'.

Reply:
It was decided long time ago that the default will be to on. It's more convenient. You should not change this, users have the option to modify this through `WithFile` option.

Reply:
@zhouya0 would you write a test that verifies these defaults?

Reply:
Yes, of course. But before I want to make few things clear:
1. `kustomize` must need `recursive` to be `false` ,  how do I support `kustomize` in `kubectl wait`?
2. If default `recursive` is `true`, should we disable this flag in `kubectl wait` since it's not said in doc or in help info?
3. The test case for recursive files seems not easy to implement, can I just add a test to check the default flags?